lib/modules/manager/npm/artifacts.ts

```diff
@@ -43,6 +49,9 @@ export async function updateArtifacts({
   // Asumming that corepack only needs to modify the package.json file in the root folder
   // As it should not be regular practice to have different package managers in different workspaces
   const pkgFileDir = upath.dirname(packageFileName);
+  const { additionalNpmrcContent } = processHostRules();
+  const npmrcContent = await getNpmrcContent(pkgFileDir);
+  await updateNpmrcContent(pkgFileDir, npmrcContent, additionalNpmrcContent);
```

Comment:
shouldn't this be right before the exec call instead? so it doesn't collide with other reads below?

Reply:
Diff filenames so it shouldn't collide

Reply:
not yet, but probably later.